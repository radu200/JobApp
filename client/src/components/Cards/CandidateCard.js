import React from "react";
import PropTypes from "prop-types";
import {makeStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import Avatar from "@material-ui/core/Avatar";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import { Link } from "react-router-dom";
import { MoreMsg, Years } from "./../../Utils/messages";
import RoomIcon from "@material-ui/icons/Room";

const useStyles = makeStyles ({
  bigAvatar: {
    margin: 10,
    width: 80,
    height: 80
  },
  root: {
    width: "100%",
  },

  breakWord: {
    wordWrap: "break-word",
    fontWeight: "bold"
  },

  aboutPage: {
    textDecoration: "none",
    color: "blue",
    "&:hover": {
      textDecoration: "underline"
    }
  },
  RoomIcon: {
    fontSize: 17
  }
});

const CandidateCard = ({  candidate, handleCandidateDetails}) => {
  const classes = useStyles()
  return candidate.map(candidate => {
    return (
      <div  onClick={() => handleCandidateDetails(candidate.userID)} key={candidate.userID}>
        <List className={classes.root}>
          <Paper >
            <ListItem>
              <Avatar
                className={classes.bigAvatar}
                alt={candidate.first_name}
                src={candidate.avatar}
              />
              <ListItemText
                primary={
                  <div>
                    <Typography>
                      {candidate.first_name} {candidate.last_name}
                    </Typography>
                    <Typography
                      className={classes.textBold}
                      color="textPrimary"
                    >
                      {candidate.category} - {candidate.total_ex_years} {Years}
                    </Typography>
                    <Typography color="textSecondary">
                      {candidate.position}
                    </Typography>
                    <Typography
                      className={classes.breakWord}
                      color="textSecondary"
                    >
                      {candidate.job_seeker_about_me}
                    </Typography>
                    <Typography color="textSecondary">
                      <RoomIcon className={classes.RoomIcon} />{" "}
                      {candidate.job_seeker_location}
                    </Typography>
                    <Typography
                      className={classes.breakWord}
                      color="textSecondary"
                    >
                      <Link
                        className={classes.aboutPage}
                        to={`/candidate-details/${candidate.userID}`}
                      >
                        {MoreMsg}
                      </Link>
                    </Typography>
                  </div>
                }
              />
            </ListItem>
          </Paper>
        </List>
      </div>
    );
  });
};

CandidateCard.propTypes = {
  classes: PropTypes.object,
  Years: PropTypes.string,

  candidate: PropTypes.arrayOf(
    PropTypes.shape({
      userID: PropTypes.number,
      first_name: PropTypes.string,
      last_name: PropTypes.string,
      category: PropTypes.string,
      total_ex_years: PropTypes.number,
      job_seeker_about_me: PropTypes.string,
      job_seeker_location: PropTypes.string
    })
  )
};
export default CandidateCard
