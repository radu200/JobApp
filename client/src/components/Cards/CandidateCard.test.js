import React from 'react';
import { shallow} from 'enzyme';
import CandidateCard from './CandidateCard'

///component render with no props
///componentn reder with props


describe('CandidateCard',() => {
    it('should render correctly with no props', () => {
        const component = shallow(<CandidateCard />)
        expect(component).toMatchSnapshot();

    })

    it('should render banner text correctly with given props', () => {
        const candidate = [{
         avatar:'avatar',
         category:'category',
         first_name:'John',
         last_name:'Doe',
         job_seeker_about_me:'about me',
         job_seeker_location:'Paris',
         total_ex_years:0,
         userID:1
        }]
       
        const component = shallow(<CandidateCard candidate={candidate} />);
        expect(component).toMatchSnapshot();
      });

})

