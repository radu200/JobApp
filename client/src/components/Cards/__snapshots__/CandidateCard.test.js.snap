// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandidateCard should render banner text correctly with given props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(WithStyles)
    candidate={
      Array [
        Object {
          "avatar": "avatar",
          "category": "category",
          "first_name": "John",
          "job_seeker_about_me": "about me",
          "job_seeker_location": "Paris",
          "last_name": "Doe",
          "total_ex_years": 0,
          "userID": 1,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "candidate": Array [
        Object {
          "avatar": "avatar",
          "category": "category",
          "first_name": "John",
          "job_seeker_about_me": "about me",
          "job_seeker_location": "Paris",
          "last_name": "Doe",
          "total_ex_years": 0,
          "userID": 1,
        },
      ],
      "classes": Object {
        "RoomIcon": "CandidateCard-RoomIcon-7",
        "aboutPage": "CandidateCard-aboutPage-6",
        "bigAvatar": "CandidateCard-bigAvatar-1",
        "hover": "CandidateCard-hover-5",
        "inline": "CandidateCard-inline-3",
        "root": "CandidateCard-root-2",
        "textAbout": "CandidateCard-textAbout-4",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "candidate": Array [
          Object {
            "avatar": "avatar",
            "category": "category",
            "first_name": "John",
            "job_seeker_about_me": "about me",
            "job_seeker_location": "Paris",
            "last_name": "Doe",
            "total_ex_years": 0,
            "userID": 1,
          },
        ],
        "classes": Object {
          "RoomIcon": "CandidateCard-RoomIcon-7",
          "aboutPage": "CandidateCard-aboutPage-6",
          "bigAvatar": "CandidateCard-bigAvatar-1",
          "hover": "CandidateCard-hover-5",
          "inline": "CandidateCard-inline-3",
          "root": "CandidateCard-root-2",
          "textAbout": "CandidateCard-textAbout-4",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`CandidateCard should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(WithStyles) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "RoomIcon": "CandidateCard-RoomIcon-7",
        "aboutPage": "CandidateCard-aboutPage-6",
        "bigAvatar": "CandidateCard-bigAvatar-1",
        "hover": "CandidateCard-hover-5",
        "inline": "CandidateCard-inline-3",
        "root": "CandidateCard-root-2",
        "textAbout": "CandidateCard-textAbout-4",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "RoomIcon": "CandidateCard-RoomIcon-7",
          "aboutPage": "CandidateCard-aboutPage-6",
          "bigAvatar": "CandidateCard-bigAvatar-1",
          "hover": "CandidateCard-hover-5",
          "inline": "CandidateCard-inline-3",
          "root": "CandidateCard-root-2",
          "textAbout": "CandidateCard-textAbout-4",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
